rules/ast/ast.go

```diff
@@ -29,7 +29,12 @@ import (
 	"github.com/prometheus/prometheus/storage/metric"
 )
 
-var stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+var (
+	stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+	queryTimeout = flag.Duration("query.timeout", 2*time.Minute, "Maximum time a query may take before being aborted.")
+)
+
+var queryTimeoutErr = fmt.Errorf("query timeout after %v", *queryTimeout)
```

Comment:
I had always just had this in the back of my head without any concrete plans for the roadmap. Now that I formulated it, it doesn't seem *too* complicated anymore ;) So while I can't promise that it'll happen soon, I think your suggestion is fair to code under the assumption that it will happen at some point... so yes, let's keep the timeouts out of the timer code for now.